runtime/runtime/src/lib.rs

```diff
@@ -1335,6 +1335,31 @@ impl Runtime {
 
         let (trie_changes, state_changes) = state_update.finalize()?;
 
+        // Precompile contracts just inserted into state and store result (compiled code or error)
+        // in the database.
+        // Note that contract compilation costs are already accounted in deploy cost using
+        // special logic in estimator (see get_runtime_config() function).
+        let wasm_config = apply_state.config.wasm_config.clone();
+        let cache = apply_state.cache.as_deref();
+        state_changes
+            .par_iter()
+            .map(|state_change| match state_change.trie_key {
+                TrieKey::ContractCode { .. } => {
+                    let code = state_change
+                        .changes
+                        .last()
+                        .expect("Committed entry should have at least one change")
+                        .data
+                        .as_ref();
+                    if let Some(code) = code {
```

Comment:
What happens if this is None and is it possible at all?

Reply:
If code is None it means that it was just removed from the state, thus there is nothing to precompile. I will add comment to clarify it